core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -166,8 +167,26 @@ private[spark] object JettyUtils extends Logging {
     contextHandler
   }
 
+  private def addDefaultFilters(handlers: Seq[ServletContextHandler], conf: SparkConf): Unit = {
+    val compression = conf.getBoolean("spark.ui.compression", true)
```

Comment:
Since Spark supports the user to add custom Filters, I'm concerned that they may add a Filter that conflicts with `GZipFilter`. So I add this configuration to disable `GZipFilter`.